### ISSUE TITLE
findmin(A; dims) ignores first value if index is 0

### ISSUE THREAD

**Issue by cmcaine** [JuliaLang/julia#38660]
Index can be zero if you're using OffsetArrays.jl or similar. Also
affects `findmax`, `argmin`, `argmax`.

If `dims` is omitted or `:` then you don't get this bug :)

Found while working on #35316 and I'll fix it there.

Reproducible example:

```julia
julia> using OffsetArrays

julia> ov = OffsetVector([-1, 1], 0:1)
2-element OffsetArray(::Vector{Int64}, 0:1) with eltype Int64 with indices 0:1:
 -1
  1

julia> findmin(ov, dims=1) # Should be ([-1], [0])
([1], [1])
```

**Comment by FedericoStra** [comment]
The problem appears to be that, given `A = ov` and `ri = Base.reduced_indices0(A, 1)` as defined in [`_findmin`](https://github.com/JuliaLang/julia/blob/master/base/reducedim.jl#L991-L1002), [`findminmax!`](https://github.com/JuliaLang/julia/blob/master/base/reducedim.jl#L907-L955) gets called with arguments
```julia
julia> Rval = fill!(similar(A, ri), first(A))
1-element OffsetArray(::Vector{Int64}, 0:0) with eltype Int64 with indices 0:0:
 -1

julia> Rind = zeros(eltype(keys(A)), ri)
1-element OffsetArray(::Vector{Int64}, 0:0) with eltype Int64 with indices 0:0:
 0
```
but, as [the comment above `findminmax!`](https://github.com/JuliaLang/julia/blob/master/base/reducedim.jl#L905) says, "The initial values of `Rval` are not used if the corresponding indices in `Rind` are `0`".



```
findminmax!(isless, fill!(similar(A, ri), first(zeros(eltype(keys(A)), ri), A)
```

**Comment by cmcaine** [comment]
Yeah, I see why it happens. I guess someone should decide if that behaviour
(ignore if Rind is zero(eltype(keys(A)))) is something that should be
retained.

I don't think it can be retained if zero is a valid index into the array,
tho maybe I can find a clever way around it.

**Comment by FedericoStra** [comment]
> maybe I can find a clever way around it

Making `Rind::Array{Union{Nothing,Int},N}` seems the most obvious and laziest thing to do in order to change the least amount of code.

**Comment by N5N3** [comment]
Maybe we can replace `zero(eltype(keys(A)))` with `prevind(A, first(keys(A)))` to indicate the state of initialization.
